libdrgn/arch_ppc64.c

```diff
@@ -267,6 +280,193 @@ apply_elf_reloc_ppc64(const struct drgn_relocating_section *relocating,
 	}
 }
 
+struct pt_level {
+	uint16_t bits;
+	uint16_t shift;
+	uint64_t entries;
+};
+
+struct pgtable_iterator_ppc64 {
+	uint16_t levels;
+	uint64_t pt_mask;
+	uint64_t page_size;
+	struct pgtable_iterator it;
+	const struct pt_level *pt_levels;
+};
+
+static const struct pt_level pt_levels_radix_4k[] = {
+	{ .bits = 9,  .shift = 12, .entries = 512  },	/* PTE */
+	{ .bits = 9,  .shift = 21, .entries = 512  },	/* PMD */
+	{ .bits = 9,  .shift = 30, .entries = 512  },	/* PUD */
+	{ .bits = 13, .shift = 39, .entries = 8192 },	/* PGD */
+};
+
+static const struct pt_level pt_levels_radix_64k[] = {
+	{ .bits = 5,  .shift = 16, .entries = 32   },	/* PTE */
+	{ .bits = 9,  .shift = 21, .entries = 512  },	/* PMD */
+	{ .bits = 9,  .shift = 30, .entries = 512  },	/* PUD */
+	{ .bits = 13, .shift = 39, .entries = 8192 },	/* PGD */
+};
+
+static inline uint64_t
+get_page_mask(struct pgtable_iterator_ppc64 *it, int level)
+{
+	return (((1UL << 53) - 1) & ~((1UL << it->pt_levels[level].shift) - 1));
```

Comment:
`UINT64_C(1)` instead of `1UL` here, too.

Reply:
Sure.